version: "3.8"

services:
  postgres:
    image: postgres
    env_file:
      - .env
    ports:
      - "5432:5432"
    healthcheck:
      test:
        [
          "CMD-SHELL",
          "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB} -p ${POSTGRES_PORT}",
        ]
      interval: 5s
      timeout: 5s
      retries: 5
    volumes:
      - dev_db:/var/lib/postgresql/data

  app:
    build:
      context: .
      dockerfile: dev.dockerfile
      args:
        - UID=${UID}
        - GID=${GID}
    ports:
      - "8000:8000"
    depends_on:
      postgres:
        condition: service_healthy
    env_file:
      - .env
    environment:
      ENV: dev
    volumes:
      - .:/zelthy/

  celery:
    build:
      context: .
      dockerfile: dev.dockerfile
      args:
        - UID=${UID}
        - GID=${GID}
    command: /bin/sh -c "cd ${PROJECT_NAME} && celery -A ${PROJECT_NAME} worker -l info"
    volumes:
      - .:/zelthy/
    env_file:
      - .env
    depends_on:
      app:
        condition: service_started
      redis:
        condition: service_healthy
        

  redis:
    image: redis
    ports:
      - "6379:6379"
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 10s
      timeout: 5s
      retries: 3

  celery_beat:
    image: kczelthy/zelthy3:latest
    command: /bin/sh -c "cd ${PROJECT_NAME} && celery -A ${PROJECT_NAME} beat -l info --scheduler django_celery_beat.schedulers:DatabaseScheduler"
    volumes:
      - .:/zelthy/
    env_file:
      - .env
    depends_on:
      app:
        condition: service_started
      redis:
        condition: service_healthy


volumes:
  dev_db:
